packages/medusa/src/services/pricing.ts

```diff
@@ -0,0 +1,333 @@
+  productVariantService: ProductVariantService
+  productService: ProductService
+  taxProviderService: TaxProviderService
+  cartService: CartService
+  regionService: RegionService
+  priceSelectionStrategy: IPriceSelectionStrategy
+}
+
+/**
+ * Allows retrieval of prices.
+ * @extends BaseService
+ */
+class PricingService extends BaseService {
+  protected manager_: EntityManager
+  protected readonly cartService: CartService
+  protected readonly regionService: RegionService
+
+  constructor({
+    manager,
+    cartService,
+    productVariantService,
+    productService,
+    taxProviderService,
+    regionService,
+    priceSelectionStrategy,
+  }: InjectedDependencies) {
+    super()
+    this.manager_ = manager
+    this.regionService = regionService
+    this.taxProviderService = taxProviderService
+    this.cartService = cartService
+    this.priceSelectionStrategy = priceSelectionStrategy
+    this.productVariantService = productVariantService
+    this.productService = productService
+  }
+
+  withTransaction(transactionManager: EntityManager): PricingService {
+    if (!transactionManager) {
+      return this
+    }
+
+    const cloned = new PricingService({
+      manager: transactionManager,
+      cartService: this.cartService,
+      productVariantService: this.productVariantService,
+      productService: this.productService,
+      taxProviderService: this.taxProviderService,
+      regionService: this.regionService,
+      priceSelectionStrategy: this.priceSelectionStrategy,
+    })
+
+    cloned.transactionManager_ = transactionManager
+
+    return cloned
+  }
+
+  async collectPricingContext(
+    context: PriceSelectionContext
+  ): Promise<PricingContext> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        let automaticTaxes = false
+        let taxRate = null
+        let regionId = context.region_id
+        let currencyCode = context.currency_code
+
+        if (context.cart_id) {
+          const cart = await this.cartService
+            .withTransaction(transactionManager)
+            .retrieve(context.cart_id, {
+              select: ["id", "region_id"],
+            })
+
+          regionId = cart.region_id
+        }
+
+        if (regionId) {
+          const region = await this.regionService
+            .withTransaction(transactionManager)
+            .retrieve(regionId, {
+              select: ["id", "currency_code", "automatic_taxes", "tax_rate"],
+            })
+
+          currencyCode = region.currency_code
+          automaticTaxes = region.automatic_taxes
+          taxRate = region.tax_rate
+        }
+
+        return {
+          price_selection: {
+            ...context,
+            region_id: regionId,
+            currency_code: currencyCode,
+          },
+          automatic_taxes: automaticTaxes,
+          tax_rate: taxRate,
+        }
+      }
+    )
+  }
+
+  async calculateTaxes(
+    prices: ProductVariantPricing,
```

Comment:
Prices lets think that it would be an array of price we should probably rename that?